apis/kueue/v1alpha1/workload_webhook_test.go

```diff
@@ -0,0 +1,80 @@
+package v1alpha1
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// makeWorkload is copied from "sigs.k8s.io/kueue/pkg/util/testing"
+// NOTE: we could not import "sigs.k8s.io/kueue/pkg/util/testing" because of cyclic import.
+func makeWorkload(override func(*Workload)) *Workload {
+	obj := &Workload{
+		ObjectMeta: metav1.ObjectMeta{Name: "name", Namespace: "ns"},
+		Spec: WorkloadSpec{
+			PodSets: []PodSet{
+				{
+					Name:  "main",
+					Count: 1,
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "c",
+								Resources: corev1.ResourceRequirements{
+									Requests: make(corev1.ResourceList),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	override(obj)
+	return obj
+}
+
+func TestValidateWorkload(t *testing.T) {
+	specField := field.NewPath("spec")
+	podSetsField := specField.Child("podSets")
+	testCases := map[string]struct {
+		workload    *Workload
+		expectedErr field.ErrorList
+	}{
+		"should have at least one podSet": {
+			workload: makeWorkload(func(w *Workload) {
+				w.Spec.PodSets = nil
+			}),
+			expectedErr: field.ErrorList{
+				field.Required(podSetsField, "at least one podSet is required"),
+			},
+		},
+		"count should be greater than 0": {
+			workload: makeWorkload(func(w *Workload) {
+				w.Spec.PodSets[0].Count = -1
+			}),
+			expectedErr: field.ErrorList{
+				field.Invalid(podSetsField.Index(0).Child("count"), int32(-1), "count must be greater than 0"),
+			},
+		},
+		"should have valid priorityClassName": {
+			workload: makeWorkload(func(w *Workload) {
+				w.Spec.PriorityClassName = "invalid_class"
+			}),
+			expectedErr: field.ErrorList{
+				field.Invalid(specField.Child("priorityClassName"), "invalid_class", "a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"),
```

Comment:
this is a lot of text, subject to change

If you use cmp.Equal, you can pass cmpopts.IgnoreFields so that we don't have to compare message.